File: server/ai.go
package main

import (
	"fmt"
	"math/rand"
)

const MAX = 0xff - 2
const (
	NORTH = iota
	EAST  = iota
	SOUTH = iota
	WEST  = iota
)

const (
	UP    = iota
	RIGHT = iota
	DOWN  = iota
	LEFT  = iota
)

type Player struct {
	x byte
	y byte
}

type Context struct {
	Seed  int
	Index int
	Count int
	Rand  rand.Rand
}

func (p Player) isDead() bool { return p.x >= MAX }

func ai(players []Player, ctx *Context) byte {
	fmt.Println(players[0])
	// always go down
	if ctx.Index%2 == 0 {
		return SOUTH
	}
	return NORTH
}

File: js/ai.js
const [ NORTH, EAST, SOUTH, WEST ] = Array(4).keys()

// context.size: the size of the map (max 250)
// context.count: the amount of players in the game
// context.index: your personal index
// context.rand: a seeded random function
exports.buildContext = context => {

  // prepare you context here

  return {
    // we return a name, but you can add anything
  }
}


const wait = t => new Promise(s => setTimeout(s, t))
// context: contain everything you returned in buildContext
// players: the list of the players coordinate, you are the first in the list
exports.ai = async (context, players) => {
  // Return a direction here, choose wisely

  // always EAST, always right
  //await wait(Math.random()*1000)
  return EAST
}

exports.name = 'wesh'
